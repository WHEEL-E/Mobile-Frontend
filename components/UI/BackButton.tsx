import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

interface InputBackButtonProps {
  color: string;
  size: number;
  onPress: () => void;
}

export const BackButton = (props: InputBackButtonProps) => {
  const { color, size, onPress } = props;

  return (
    <TouchableOpacity>
      <Ionicons name="arrow-back" color={color} size={size} onPress={onPress} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({});
