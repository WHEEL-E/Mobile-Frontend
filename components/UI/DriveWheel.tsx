import { Ionicons } from "@expo/vector-icons";
import { LinearGradient } from "expo-linear-gradient";
import { StyleSheet, TouchableOpacity, View } from "react-native";
import colors from "../../constants/colors";
import { DEVICE_WIDTH } from "../../constants/dimentions";

const DriveWheel = () => {
  return (
    <View style={styles.background}>
      <LinearGradient
        colors={["#000000", "#494949", "#000000"]}
        style={styles.linearGradient}
      >
        <TouchableOpacity style={styles.arrow}>
          <Ionicons name="chevron-up-outline" color="white" size={30} />
        </TouchableOpacity>
        <View style={styles.sideArrowsContainer}>
          <TouchableOpacity style={styles.arrow}>
            <Ionicons name="chevron-back-outline" color="white" size={30} />
          </TouchableOpacity>
          <LinearGradient
            colors={["#000000", "#000000", "#949494"]}
            style={styles.secondLinearGradient}
          >
            <LinearGradient
              colors={["#949494", "#000000", "#000000"]}
              style={styles.ThirdLinearGradient}
            >
              <LinearGradient
                colors={["#949494", "#000000", "#000000", "#949494"]}
                style={styles.fourthLinearGradient}
              >
                <View style={styles.whiteDot} />
              </LinearGradient>
            </LinearGradient>
          </LinearGradient>
          <TouchableOpacity style={styles.arrow}>
            <Ionicons name="chevron-forward-outline" color="white" size={30} />
          </TouchableOpacity>
        </View>
        <TouchableOpacity style={styles.arrow}>
          <Ionicons name="chevron-down-outline" color="white" size={30} />
        </TouchableOpacity>
      </LinearGradient>
    </View>
  );
};

const styles = StyleSheet.create({
  arrow: {
    alignSelf: "center",
  },
  sideArrowsContainer: {
    flexDirection: "row",
  },
  background: {
    borderRadius: DEVICE_WIDTH * 0.45,
    width: DEVICE_WIDTH * 0.9,
    height: DEVICE_WIDTH * 0.9,
    backgroundColor: colors.lightPurple,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 50,
    marginBottom: 30,
  },
  linearGradient: {
    borderRadius: DEVICE_WIDTH * 0.4,
    width: DEVICE_WIDTH * 0.8,
    height: DEVICE_WIDTH * 0.8,
    justifyContent: "center",
    alignItems: "center",
  },
  secondLinearGradient: {
    borderRadius: DEVICE_WIDTH * 0.3,
    width: DEVICE_WIDTH * 0.6,
    height: DEVICE_WIDTH * 0.6,
    justifyContent: "center",
    alignItems: "center",
  },
  ThirdLinearGradient: {
    borderRadius: DEVICE_WIDTH * 0.2,
    width: DEVICE_WIDTH * 0.4,
    height: DEVICE_WIDTH * 0.4,
    justifyContent: "center",
    alignItems: "center",
  },
  fourthLinearGradient: {
    borderRadius: DEVICE_WIDTH * 0.1,
    width: DEVICE_WIDTH * 0.2,
    height: DEVICE_WIDTH * 0.2,
    justifyContent: "center",
    alignItems: "center",
  },
  whiteDot: {
    borderRadius: DEVICE_WIDTH * 0.0125,
    width: DEVICE_WIDTH * 0.025,
    height: DEVICE_WIDTH * 0.025,
    backgroundColor: "white",
  },
});

export default DriveWheel;
