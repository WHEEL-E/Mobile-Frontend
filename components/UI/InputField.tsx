import React from "react";
import { StyleSheet, TextInput } from "react-native";
import colors from "../../constants/colors";
import { AutoCompleteType } from "../../utilities/types/AutoCompleteType";
interface InputFieldProps {
  placeHolder: string;
  fieldStyle: object;
  autoComplete: AutoCompleteType;
  value: string;
  onChangeText: (text: string) => void;
}

const InputField = (props: InputFieldProps) => {
  const { placeHolder, fieldStyle, autoComplete, value, onChangeText } = props;

  return (
    <TextInput
      placeholder={placeHolder}
      style={{ ...styles.input, ...fieldStyle }}
      autoCompleteType={autoComplete}
      value={value}
      onChangeText={onChangeText}
    />
  );
};

export default InputField;

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    height: 55,
    backgroundColor: "#ccc",
    textAlign: "center",
  },
  input: {
    borderRadius: 15,
    textAlign: "left",
    padding: 20,
    backgroundColor: colors.lightGray,
  },
});
