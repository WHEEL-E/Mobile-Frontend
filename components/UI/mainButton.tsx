import React from "react";
import { StyleSheet, View, TouchableOpacity, Text, Image } from "react-native";
import { Ionicons } from "@expo/vector-icons";

interface MainButtonProps {
  title: string;
  buttonStyle: object;
  titleStyle: object;
  onPress: () => void;
  icon: { name: any; size: number; color: string };
  image: { url: any };
  hasIcon: boolean;
}

export const MainButton = (props: MainButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress, icon, image } = props;

  if (props.hasIcon) {
    return (
      <TouchableOpacity
        style={{ ...buttonStyle, ...styles.button }}
        onPress={onPress}
      >
        <View style={styles.imageContainer}>
          <Ionicons name={icon.name} size={icon.size} color={icon.color} />
        </View>
        <Text style={titleStyle}>{title}</Text>
      </TouchableOpacity>
    );
  }
  return (
    <TouchableOpacity
      style={{ ...buttonStyle, ...styles.button }}
      onPress={onPress}
    >
      <View style={styles.imageContainer}>
        <Image source={image.url} style={{ ...styles.image }} />
      </View>
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
  },
  imageContainer: {
    width: "60%",
    height: "60%",
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "center",
  },
});
