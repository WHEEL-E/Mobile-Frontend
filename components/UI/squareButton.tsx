import React from "react";
import {
  StyleSheet,
  View,
  TouchableNativeFeedback,
  TouchableOpacity,
  Platform,
  Text,
  Image,
} from "react-native";

interface SquareButtonProps {
  title: string;
  buttonStyle: object;
  titleStyle: object;
  onPress: () => void;
}

export const SquareButton = (props: SquareButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress } = props;

  return (
    <TouchableOpacity
      style={{ ...buttonStyle, ...styles.button }}
      onPress={onPress}
    >
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 2,
    borderColor: "white",
    borderRadius: 10,
  },
  imageContainer: {
    flex: 1,
    width: 100,
    height: 100,
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "stretch",
  },
});
