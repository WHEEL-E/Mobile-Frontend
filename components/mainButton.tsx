import React from "react";
import {
  StyleSheet,
  View,
  TouchableNativeFeedback,
  TouchableOpacity,
  Platform,
  Text,
  Image,
} from "react-native";
import { Ionicons } from "@expo/vector-icons";

interface MainButtonProps {
  title: string;
  buttonStyle: object;
  containerStyle: object;
  titleStyle: object;
  onPress: any;
  icon: { name: any; size: number; color: string };
  image: { url: any };
}

const MainButton = (props: MainButtonProps) => {
  const {
    title,
    buttonStyle,
    containerStyle,
    titleStyle,
    onPress,
    icon,
    image,
  } = props;
  // search for a better practice for the type
  let TouchCmp: any = TouchableOpacity;
  if (Platform.OS === "android" && Platform.Version >= 21) {
    TouchCmp = TouchableNativeFeedback;
  }
  //   <Ionicons name={icon.name} size={icon.size} color={icon.color} />
  return (
    <View style={containerStyle}>
      <TouchCmp style={{ ...buttonStyle, ...styles.button }} onPress={onPress}>
        <View style={styles.imageContainer}>
          <Image source={image.url} style={styles.image} />
        </View>
        <Text style={titleStyle}>{title}</Text>
      </TouchCmp>
    </View>
  );
};

export default MainButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
  },
  imageContainer: {
    flex: 1,
    width: 100,
    height: 100,
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "stretch",
  },
});
