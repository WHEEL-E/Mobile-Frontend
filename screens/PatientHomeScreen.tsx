import React from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import { useTranslation } from "react-i18next";
import colors from "../utilities/constants/colors";
import fonts from "../utilities/constants/fonts";
import { MainButton } from "../components/buttons/MainButton";
import { SquareButton } from "../components/buttons/SquareButton";
import { PatientHomeProps } from "../utilities/types/navigationTypes/mainNavigationTypes";
import { EmergencyCallModal } from "../components/homeScreenComponents/EmergencyCallModal";

const PatientHomeScreen = (props: PatientHomeProps) => {
  const [isModalVisible, setIsModalVisible] = React.useState(false);
  const { navigation } = props;
  const { t } = useTranslation();

  return (
    <View style={styles.container}>
      <EmergencyCallModal
        isModalVisible={isModalVisible}
        setIsModalVisible={setIsModalVisible}
      />
      <View>
        <Image
          style={styles.logo}
          source={require("../assets/images/logo-b-app.png")}
        />
        <Text style={styles.mainText}>
          {t("patientHomeScreen.morning")}user
        </Text>
        <Text style={styles.subText} testID="welcomeText">
          {t("patientHomeScreen.wishGoodDay")}
        </Text>
      </View>
      <View style={styles.buttons}>
        <View style={styles.buttonsCol}>
          <MainButton
            title={t("patientHomeScreen.map")}
            titleStyle={{ ...styles.bigButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.mainButton,
              backgroundColor: colors.darkGreen,
            }}
            onPress={() => {}}
            image={{ url: require("../assets/images/map.png") }}
          />
          <SquareButton
            title={t("patientHomeScreen.emergencyCall")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.darkPink,
            }}
            onPress={() => {
              setIsModalVisible(true);
            }}
          />
          <SquareButton
            title={t("patientHomeScreen.freeDriving")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.darkGreen,
            }}
            onPress={() => {
              navigation.navigate("FreeDrive");
            }}
          />
          <SquareButton
            title={t("patientHomeScreen.addSupervisor")}
            titleStyle={{ ...styles.smallButtonTitle, color: colors.darkBlue }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.lightPurple,
            }}
            onPress={() => {}}
          />
          <SquareButton
            title={t("patientHomeScreen.help")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.lightGreen,
            }}
            onPress={() => {}}
          />
        </View>
        <View style={styles.buttonsCol}>
          <MainButton
            title={t("patientHomeScreen.healthStatus")}
            titleStyle={{ ...styles.bigButtonTitle, color: colors.darkBlue }}
            buttonStyle={{
              ...styles.mainButton,
              backgroundColor: colors.lightPurple,
            }}
            onPress={() => {}}
            image={{ url: require("../assets/images/health-state.png") }}
          />
          <SquareButton
            title={t("patientHomeScreen.getChair")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.darkGreen,
            }}
            onPress={() => {}}
          />
          <SquareButton
            title={t("patientHomeScreen.reminders")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.lightGreen,
            }}
            onPress={() => {
              navigation.navigate("Reminders");
            }}
          />
          <MainButton
            title={t("patientHomeScreen.myProfile")}
            titleStyle={styles.profileTitle}
            buttonStyle={styles.myProfileButton}
            onPress={() => {
              navigation.navigate("Profile");
            }}
            icon={{
              name: "person-outline",
              size: 70,
              color: colors.lightBrown,
            }}
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white",
    paddingHorizontal: 20,
    paddingTop: "10%",
  },
  logo: {
    height: 40,
    resizeMode: "center",
    width: "70%",
    alignSelf: "center",
  },
  mainText: {
    fontFamily: fonts.CairoBold,
    fontSize: 30,
  },
  subText: {
    fontFamily: fonts.CairoMedium,
    color: colors.darkGrey,
    fontSize: 20,
  },
  buttons: {
    marginBottom: 100,
    flex: 1,
    flexDirection: "row",
  },
  buttonsCol: {
    flex: 1,
  },
  mainButton: {
    borderRadius: 30,
    justifyContent: "center",
    flex: 3,
    margin: 2,
  },
  smallButton: {
    flex: 1,
    backgroundColor: colors.darkGreen,
    margin: 2,
  },
  myProfileButton: {
    backgroundColor: colors.darkGreen,
    justifyContent: "center",
    flex: 2,
    borderRadius: 30,
    margin: 2,
  },
  bigButtonTitle: {
    fontFamily: fonts.CairoBold,
    fontSize: 30,
    lineHeight: 40,
  },
  smallButtonTitle: {
    fontFamily: fonts.CairoSemiBold,
    fontSize: 18,
  },
  profileTitle: {
    fontFamily: fonts.CairoSemiBold,
    fontSize: 20,
    color: "white",
  },
});

export default PatientHomeScreen;
