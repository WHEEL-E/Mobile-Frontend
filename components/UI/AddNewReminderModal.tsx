import React from "react";
import { StyleSheet, Modal, Text, View } from "react-native";
import { SquareButton } from "../UI/squareButton";
import InputField from "../UI/InputField";
import colors from "../../constants/colors";

interface addNewReminderModalProps {
  modalVisible: boolean;
  setModalVisible: (modalVisible: boolean) => void;
  reminderTitleEdit: string;
  editTitleHandler: (title: string) => void;
  reminderBodyEdit: string;
  editBodyHandler: (body: string) => void;
  addNewReminderHandler: (
    reminderTitleEdit: string,
    reminderBodyEdit: string
  ) => void;
}

const AddNewReminderModal = (props: addNewReminderModalProps) => {
  const {
    modalVisible,
    setModalVisible,
    reminderTitleEdit,
    editTitleHandler,
    reminderBodyEdit,
    editBodyHandler,
    addNewReminderHandler,
  } = props;

  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={modalVisible}
      onRequestClose={() => {
        setModalVisible(!modalVisible);
      }}
    >
      <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <Text style={styles.modalTitle}>
            Type any information you need to remind your patient with
          </Text>
          <InputField
            placeHolder="Enter Reminder Title"
            value={reminderTitleEdit}
            onChangeText={editTitleHandler}
            fieldStyle={{ width: "100%", marginBottom: 10 }}
            autoComplete="off"
          />
          <InputField
            placeHolder="Enter Reminder Description"
            value={reminderBodyEdit}
            onChangeText={editBodyHandler}
            fieldStyle={{ width: "100%", height: 100 }}
            autoComplete="off"
          />
          <View style={styles.buttonsList}>
            <SquareButton
              title="cancel"
              titleStyle={{ color: "#fff" }}
              onPress={() => setModalVisible(!modalVisible)}
              buttonStyle={styles.cancelButton}
            />
            <SquareButton
              title="send"
              titleStyle={{ color: "#fff" }}
              onPress={() =>
                addNewReminderHandler(reminderTitleEdit, reminderBodyEdit)
              }
              buttonStyle={styles.sendButton}
            />
          </View>
        </View>
      </View>
    </Modal>
  );
};

export default AddNewReminderModal;

const styles = StyleSheet.create({
  modalTitle: {
    fontFamily: "Cairo-SemiBold",
    fontSize: 18,
    lineHeight: 22,
    textAlign: "center",
    marginBottom: 15,
  },
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    width: 320,
  },
  cancelButton: {
    backgroundColor: colors.darkPink,
    width: 100,
    height: 50,
  },
  sendButton: {
    backgroundColor: colors.lightGreen,
    width: 100,
    height: 50,
  },
  buttonsList: {
    flexDirection: "row",
    justifyContent: "space-around",
    marginTop: 25,
  },
});
