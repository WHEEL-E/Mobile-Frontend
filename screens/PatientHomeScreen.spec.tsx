import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import PatientHomeScreen from "./PatientHomeScreen";
import * as mainButton from "../components/UI/mainButton";
import * as squareButton from "../components/UI/squareButton";
import { MainButton } from "../components/UI/mainButton";
import { SquareButton } from "../components/UI/squareButton";

describe("patienthomeScreen", () => {
  it("makes sure screen matches snapShot", () => {
    const dummyProp: any = () => {};
    const PaitnethomeScreen = renderer
      .create(<PatientHomeScreen navigation={dummyProp} route={dummyProp} />)
      .toJSON();
    expect(PaitnethomeScreen).toMatchSnapshot();
  });

  it("makes sure button componenets are calledd", () => {
    jest.spyOn(mainButton, "MainButton");
    jest.spyOn(squareButton, "SquareButton");

    const dummyProp: any = () => {};
    render(<PatientHomeScreen navigation={dummyProp} route={dummyProp} />);
    expect(MainButton).toBeCalledTimes(3);
    expect(SquareButton).toBeCalledTimes(6);
  });

  it("makes sure text is rendered properly", () => {
    const dummyProp: any = () => {};
    const { getByTestId } = render(
      <PatientHomeScreen navigation={dummyProp} route={dummyProp} />
    );

    const welcomeText = getByTestId("welcomeText");
    expect(welcomeText.children).toContain("We wish you have a good day");
  });
});
