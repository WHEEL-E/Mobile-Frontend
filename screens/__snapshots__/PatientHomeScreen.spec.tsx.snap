// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patienthomeScreen makes sure screen matches snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingTop": "10%",
    }
  }
>
  <View>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 40,
          "resizeMode": "center",
          "width": "70%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "Cairo-Bold",
          "fontSize": 30,
        }
      }
    >
      Good morning, user
    </Text>
    <Text
      style={
        Object {
          "color": "#8F9198",
          "fontFamily": "Cairo-Medium",
          "fontSize": 20,
        }
      }
      testID="welcomeText"
    >
      We wish you have a good day
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 100,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderRadius": 30,
            "flex": 3,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "40%",
              "justifyContent": "center",
              "width": "40%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-Bold",
              "fontSize": 30,
              "lineHeight": 40,
            }
          }
        >
          Map
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#B91646",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          Emergency Call
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          Free Driving
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7D9EA",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#19456B",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          Add a supervisor
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#16C79A",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          Help
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7D9EA",
            "borderRadius": 30,
            "flex": 3,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "40%",
              "justifyContent": "center",
              "width": "40%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#19456B",
              "fontFamily": "Cairo-Bold",
              "fontSize": 30,
              "lineHeight": 40,
            }
          }
        >
          Health status
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          Get my chair
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#16C79A",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          Reminders
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderRadius": 30,
            "flex": 2,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "40%",
              "justifyContent": "center",
              "width": "40%",
            }
          }
        >
          <Text />
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 20,
            }
          }
        >
          My profile
        </Text>
      </View>
    </View>
  </View>
</View>
`;
